### PR TITLE
dev/win32 minor fixes

### PR DESCRIPTION
**Pull request by zerhacken**
- fix uv_handle_size and uv_req_size when failing return 0
- fix VC++ warning C4244
- fix VC++ warning C4389

### CODE REVIEW

**Comment by zerhacken** [comment]
@Nodejs-Jenkins signed CLA

File: src/uv-common.c [hunk 1]
```diff
@@ -52,7 +52,7 @@ size_t uv_handle_size(uv_handle_type type) {
   switch (type) {
     UV_HANDLE_TYPE_MAP(XX)
     default:
-      return -1;
+      return 0;
```
Comment (retep998) [inline src/uv-common.c:55]: Wouldn't it be better to return (size_t)-1 here?

**Comment by zerhacken** [comment]
@retep998 thanks, you are right. Will fix it.

File: src/uv-common.c [hunk 2]
```diff
@@ -52,7 +52,7 @@ size_t uv_handle_size(uv_handle_type type) {
   switch (type) {
     UV_HANDLE_TYPE_MAP(XX)
     default:
-      return -1;
+      return (size_t) - 1;
```
Comment (saghul) [inline src/uv-common.c]: This is not a good idea, it will return the biggest possible integer, which is not what we want.

**Comment by zerhacken** [comment]
Revert to my original fix returning 0?

**Comment by saghul** [comment]
I guess so, yes. The size of a handle needs to be > 0.

**Comment by zerhacken** [comment]
@saghul reverted to original fix, hope this is better?

File: src/uv-common.c [hunk 3]
```diff
@@ -101,7 +101,7 @@ const char* uv_strerror(int err) {
 int uv_ip4_addr(const char* ip, int port, struct sockaddr_in* addr) {
   memset(addr, 0, sizeof(*addr));
   addr->sin_family = AF_INET;
-  addr->sin_port = htons(port);
+  addr->sin_port = htons((u_short) port);
```
Comment (saghul) [inline src/uv-common.c]: I'm not seeing any warning here, and u_short is a Windows-ism. uint16_t would be fine but I've never seen a warning, what system are you using?

File: src/uv-common.c [hunk 4]
```diff
@@ -115,7 +115,7 @@ int uv_ip6_addr(const char* ip, int port, struct sockaddr_in6* addr) {
 
   memset(addr, 0, sizeof(*addr));
   addr->sin6_family = AF_INET6;
-  addr->sin6_port = htons(port);
+  addr->sin6_port = htons((u_short) port);
```
Comment (saghul) [inline src/uv-common.c]: Ditto.

File: src/win/poll.c [hunk 5]
```diff
@@ -209,7 +209,7 @@ static int uv__fast_poll_set(uv_loop_t* loop, uv_poll_t* handle, int events) {
   assert(!(handle->flags & UV__HANDLE_CLOSING));
   assert((events & ~(UV_READABLE | UV_WRITABLE)) == 0);
 
-  handle->events = events;
+  handle->events = (unsigned char) events;
```
Comment (saghul) [inline src/win/poll.c]: Maybe we should declare events as unsigned int instead.

File: src/win/poll.c [hunk 6]
```diff
@@ -451,7 +451,7 @@ static int uv__slow_poll_set(uv_loop_t* loop, uv_poll_t* handle, int events) {
...
```
Comment (saghul) [inline src/win/poll.c:454]: Ditto.

File: src/uv-common.c (hunk 3 above)
Comment (zerhacken) [inline src/uv-common.c]: VC++ (2012 + 2013) its a /L4 warning, you are right uint16_t would be better.

Comment (saghul) [inline src/uv-common.c]: What is a /L4 warning?

Also, the function itself accepts an int, but ports cannot be negative, so maybe we need to adjust uv_ip4_add itself...

File: src/win/poll.c (hunk 5 above)
Comment (zerhacken) [inline src/win/poll.c]: That would be perfect.

File: src/uv-common.c (hunk 3 above)
Comment (zerhacken) [inline src/uv-common.c]: /L4 short hand for compiler warning level 4 (http://msdn.microsoft.com/en-us/library/thxezb7y.aspx).

Will look into adjusting uv_ip4_add to take an unsigned int if you are fine with that?
Actually "uv_ip4_add" might as well take an unsigned short?

**Comment by zerhacken** [comment]
@saghul this I like better, should be warning-less and even a tad prettier

**Comment by zerhacken** [comment]
The test failure on smartos build doesnt appear as related

**Pull request closed without merging**